src/main/java/io/arlas/server/rest/tag/UpdateServices.java

```diff
@@ -0,0 +1,147 @@
+ * this work for additional information regarding copyright
+ * ownership. Gisaïa licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.arlas.server.rest.tag;
+
+import io.arlas.server.app.ArlasServerConfiguration;
+import io.arlas.server.core.FluidSearch;
+import io.arlas.server.exceptions.ArlasException;
+import io.arlas.server.exceptions.BadRequestException;
+import io.arlas.server.exceptions.NotImplementedException;
+import io.arlas.server.model.CollectionReference;
+import io.arlas.server.model.request.MixedRequest;
+import io.arlas.server.model.request.Search;
+import io.arlas.server.model.request.Tag;
+import io.arlas.server.model.response.UpdateResponse;
+import io.arlas.server.rest.explore.ExploreServices;
+import org.elasticsearch.client.Client;
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.index.reindex.BulkByScrollResponse;
+import org.elasticsearch.index.reindex.UpdateByQueryAction;
+import org.elasticsearch.index.reindex.UpdateByQueryRequestBuilder;
+import org.elasticsearch.script.Script;
+import org.elasticsearch.script.ScriptType;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class UpdateServices extends ExploreServices{
+    public enum ACTION {
+        ADD,REMOVE, REMOVEALL;
+    }
+
+    public UpdateServices(Client client, ArlasServerConfiguration configuration) {
+        super(client, configuration);
+    }
+
+    public UpdateResponse tag(CollectionReference collectionReference, MixedRequest request, Tag tag, int max_updates) throws IOException, ArlasException {
+        return doAction(ACTION.ADD,collectionReference, request, tag, max_updates);
+    }
+
+    public UpdateResponse unTag(CollectionReference collectionReference, MixedRequest request, Tag tag, int max_updates) throws IOException, ArlasException {
+        return doAction(ACTION.REMOVE,collectionReference, request, tag, max_updates);
+    }
+
+    public UpdateResponse removeAll(CollectionReference collectionReference, MixedRequest request, Tag tag, int max_updates) throws IOException, ArlasException {
+        return doAction(ACTION.REMOVEALL,collectionReference, request, tag, max_updates);
+    }
+
+    protected UpdateResponse doAction(ACTION action, CollectionReference collectionReference, MixedRequest request, Tag tag, int max_updates) throws IOException, ArlasException {
```

Comment:
In order to prepare future refactoring, the elastic related code below should be in a separated class so that we could easily provide an interface and additional implementations (Mongo, Cassandra, ...).

Reply:
this requirements comes late, but ok.

Reply:
It's not a requirement, if you do not have time, we will handle it later, no pb.

Reply:
I refactored.